client/src/modules/MentorRegistry/components/MentorRegistryTableContainer.tsx

```diff
@@ -0,0 +1,327 @@
+import { useMemo, useState } from 'react';
+import { GithubUserLink } from 'components/GithubUserLink';
+import { SafetyCertificateTwoTone } from '@ant-design/icons';
+import { colorTagRenderer, getColumnSearchProps, stringSorter, tagsRenderer, dateSorter } from 'components/Table';
+import { formatDate } from 'services/formatter';
+import { Course } from 'services/models';
+import CopyToClipboardButton from 'components/CopyToClipboardButton';
+import {
+  mentorRegistryStyles,
+  MentorRegistryTabsMode,
+  MentorsRegistryColumnKey,
+  MentorsRegistryColumnName,
+  TABS,
+  TabsMode,
+} from '../constants';
+import { FilterValue } from 'antd/lib/table/interface';
+import { Button } from 'antd';
+import { DisciplineDto, MentorRegistryDto, UpdateUserDtoLanguagesEnum } from 'api';
+
+type ChildrenProp = {
+  tagFilters: string[];
+  filteredData: MentorRegistryDto[];
+  columns: any[];
+  handleTagClose: (tag: string) => void;
+  handleClearAllButtonClick: () => void;
+  handleTableChange: (_: any, filters: Record<MentorsRegistryColumnKey, FilterValue | string[] | null>) => void;
+};
+
+type Props = {
+  data: MentorRegistryDto[];
+  courses: Course[];
+  activeTab: MentorRegistryTabsMode;
+  handleModalDataChange: (mode: string, record: any) => void;
+  children: (props: ChildrenProp) => JSX.Element;
+  disciplines: DisciplineDto[];
+};
+
+export interface CombinedFilter {
```

Comment:
curious, why do you use `interface` and `type` in similar cases? ie why not to stick to one?

Reply:
fixed